exchange.c

```diff
@@ -233,14 +236,20 @@ static int domain_exchange_once(int (*layoutfunc)(int p), int** toGo_arr, int **
         endrun(787878, "Task=%d NumPart=%d All.MaxPart=%d\n", ThisTask, NumPart, All.MaxPart);
     }
 
-    if(N_sph_slots > All.MaxPart)
-        endrun(787878, "Task=%d N_sph=%d All.MaxPart=%d\n", ThisTask, N_sph_slots, All.MaxPart);
-
-    if(N_bh_slots > All.MaxPartBh)
-        endrun(787878, "Task=%d N_bh=%d All.MaxPartBh=%d\n", ThisTask, N_bh_slots, All.MaxPartBh);
-
-    if(N_star_slots > All.MaxPartBh)
-        endrun(787878, "Task=%d N_star=%d All.MaxPartBh=%d\n", ThisTask, N_star_slots, All.MaxPartBh);
+    int newSph = All.MaxPartSph;
```

Comment:
An array will really help -- the sizeof() of each element data type can be stored in an array to simplify the interface of realloc_secondary_data too.

Reply:
Note that the particle number increase is slightly different for different particles, so we still need a nest of if statements.